src/core/connection.c

```diff
@@ -3076,6 +3076,35 @@ QuicConnQueueUnreachable(
     }
 }
 
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+QuicConnQueueRouteCompletion(
+    _In_ QUIC_CONNECTION* Connection,
+    _When_(Succeeded == FALSE, _In_opt_)
+    _When_(Succeeded == TRUE, _In_)
+        const uint8_t* PhysicalAddress,
+    _In_ BOOLEAN Succeeded
+    )
+{
+    QUIC_OPERATION* ConnOper =
+        QuicOperationAlloc(Connection->Worker, QUIC_OPER_TYPE_ROUTE_COMPLETION);
+    if (ConnOper != NULL) {
+        ConnOper->ROUTE.Succeeded = Succeeded;
+        if (Succeeded) {
+            memcpy(ConnOper->ROUTE.PhysicalAddress, PhysicalAddress, sizeof(ConnOper->ROUTE.PhysicalAddress));
+        }
+        QuicConnQueueOper(Connection, ConnOper);
+    } else {
+        QuicTraceEvent(
+            AllocFailure,
+            "Allocation of '%s' failed. (%llu bytes)",
+            "Route completion operation",
+            0);
```

Comment:
We're permanently stuck if we hit this, right? Nothing will "kick" the connection to make progress? What then?

Reply:
I actually don't know what to do in this case because it takes an queue operation to kill a connection. In TCP, I'd just call the connection inline.

Reply:
But that's not a fatal issue. If you fail to get the ICMP error, you can still move forward. This is a completion call that fails. I have an idea on how to fix this.